### ISSUE TITLE
Remove bogus `null` arguments from emit(event, ...args)

### ISSUE THREAD

**Issue by rwaldron** [rwaldron/johnny-five#561]
Per discussion in https://github.com/rwaldron/johnny-five/issues/553

**Comment by rwaldron** [comment]
@dtex @BrianGenisio 

I started working on this and quickly discovered that I'm to blame for a lot of inconsistency :( Getting rid of the bogus error arg is easy, but then there's another issue: the actual data arg—this is all over the place. I'll do the removal in one patch, but we need to have a conversation about the data arg.

**Comment by rwaldron** [comment]
Also, once this lands, we'll be v0.9.0-beta

**Comment by rwaldron** [comment]
https://github.com/rwaldron/johnny-five/issues/569

**Comment by rwaldron** [comment]
Affects: 

  - Accelerometer
  - Gyro
  - Ir
  - Joystick
  - Motor
  - Ping
  - Pir
  - ReflectanceArray
  - Sensor
  - Sonar
  - Temperature
  - Wii

**Comment by scottgonzalez** [comment]
Since there doesn't seem to be an existing reference to the PR: #667.

**Comment by rwaldron** [comment]
Nothing wrong with explicit link, but Github auto links it: 

![](http://i.gyazo.com/24fb6c74b921c28f5cd2dbfa04bb44c4.png)
